### ISSUE TITLE
Removal of IEC related symbols from S101 PC

### ISSUE THREAD

**Issue by alvarosanuy** [S-101-Portrayal-subWG/Working-Documents#155]
**Decisions made at Portrayal subWG meeting on 19/10/23**

CLOSE this Issue and create  anew one to track the progress of the S101PT proposal to IEC.

_Originally posted by @alvarosanuy in https://github.com/S-101-Portrayal-subWG/Working-Documents/issues/126#issuecomment-1776140669_

**Comment by alvarosanuy** [comment]
Currently awaiting response from IEC on S101PT recommendation to centrally managing these symbols in the IHO GI Registry and making this repository the point of truth for all OEMs.

Irrespective of their decision on the S101PT recommendation, **it was decided to remove all Mariners and Manufacturers symbols from S101 PC 2.0.0.** The only exception being INDHLT and DNGHLT which are required for A&I.

Note that, based on decisions made at S101PT11, these symbols will be retained in PC 1.2.0.

- [x] @HannuPeiponen - Notify IEC that S101PC will not contain Mariners or Manufactures symbols and that will be up to OEMs to implement them in ECDIS .

**Comment by alvarosanuy** [comment]
Dear Chair of the IHO S-101PT,

 

IEC TC80/MT5 for maintenance of the IEC 62288 completed today a 2-day meeting. Please find attached liaison letter from IEC TC80/MT5 to IHO S-101PT as response to your liaison letter to the IEC.

 

Best regards

Hannu
[MT5-2309 Liaison letter from IEC TC80-MT5 to IHO S-101PT.zip](https://github.com/S-101-Portrayal-subWG/Working-Documents/files/13733318/MT5-2309.Liaison.letter.from.IEC.TC80-MT5.to.IHO.S-101PT.zip)

**Comment by alvarosanuy** [comment]
My understanding of this letter is that OEM's will source the 'Mariners' and 'Manufacturers' symbols from the S-52 PL (until a day when S-57 is sunset). 

- **Does this mean OEMs do not need these symbols 'repeated' in S-101 PC?** 

Also, IEC manifested interest in taking over all these symbols at the right time and via different mechanisms, maybe before S-57 sunset date.

**Comment by HannuPeiponen** [comment]
My reply to the question of Alvaro is: Yes, my understanding is that OEMs do not need these symbols “repeated” in S-101 PC as they can access these symbols from the S-52 PL until (and maybe even beyond) a day when S-57 is sunset.

**Comment by alvarosanuy** [comment]
**Decisions made at Portrayal subWG meeting on 09/04/24**

- [x] NIWC to list the Features that would be removed from the S-101 PC
- [ ] Alvaro to establish a short review period - No comments meaning endorsement.
- [ ] NIWC to delete the identified features from S-101 PC 1.3.0

**Comment by DavidGrant-NIWC** [comment]
|AreaFill|Description|
|:---|:---|
|OVERSC01|Overscale part of a display containing data from more than one navigation purpose|
---
|LineStyle|Description|
|:---|:---|
|CHCRDEL1|manual deletion|
|CHCRID01|manual update|
|ERBLNA01|Electronic Bearing Line|
|INDHLT02|mariners indication highlight|
|NONHODAT|boundary of non-HO data|
|SCLBDY51|chart scale boundary|
---
|Symbol|Description|
|:---|:---|
|BLKADJ01|symbol for checking brightness and contrast|
|CHCRDEL1|manually deleted or modified|
|CHCRID01|manuallly updated|
|CHINFO08|mariners information note|
|CHINFO09|mariners caution note|
|CHINFO10|manufacturers information note|
|CHINFO11|manufacturers caution note|
|CHKSYM01|symbol for checking symbol sizes|
|CURSRA01|ordinary cursor|
|CURSRB01|cursor with open centre|
|DNGHILIT|transparent danger highlight for mariner's use|
|EBLVRM11|point of origin for an offset EBL or VRM|
|ERBLTIK1|range mark for an ERBL|
|INDHLT01|indication highlight for mariner's use|
|NORTHAR1|north arrow|
|REFPNT02|reference point, 'ghost cursor' (user interface)|
|SCALEB10|one mile scale bar|
|SCALEB11|10 mile latitude bar|
|TIDCUR01|predicted tidal stream or current direction|
|TIDCUR02|actual tidal stream or current direction|
|TIDCUR03|box for current strength|

**Comment by alvarosanuy** [comment]
Should COLOUR **ADINF** be also considered for removal from S101PC ?

**Comment by HannuPeiponen** [comment]
> AreaFill	Description
> OVERSC01	Overscale part of a display containing data from more than one navigation purpose
> LineStyle	Description
> CHCRDEL1	manual deletion
> CHCRID01	manual update
> ERBLNA01	Electronic Bearing Line
> INDHLT02	mariners indication highlight
> NONHODAT	boundary of non-HO data
> SCLBDY51	chart scale boundary
> Symbol	Description
> BLKADJ01	symbol for checking brightness and contrast
> CHCRDEL1	manually deleted or modified
> CHCRID01	manuallly updated
> CHINFO08	mariners information note
> CHINFO09	mariners caution note
> CHINFO10	manufacturers information note
> CHINFO11	manufacturers caution note
> CHKSYM01	symbol for checking symbol sizes
> CURSRA01	ordinary cursor
> CURSRB01	cursor with open centre
> DNGHILIT	transparent danger highlight for mariner's use
> EBLVRM11	point of origin for an offset EBL or VRM
> ERBLTIK1	range mark for an ERBL
> INDHLT01	indication highlight for mariner's use
> NORTHAR1	north arrow
> REFPNT02	reference point, 'ghost cursor' (user interface)
> SCALEB10	one mile scale bar
> SCALEB11	10 mile latitude bar
> TIDCUR01	predicted tidal stream or current direction
> TIDCUR02	actual tidal stream or current direction
> TIDCUR03	box for current strength

I have noted that Jonathan Pritchard has created S-101 version of the "ECDIS Chart 1". Some of these sysmbols are used in "ECDIS Chart 1", therefore, such symbols should stay in PC of S-101 as the machine-readable S-100 compliant system needs them for rendering of the S-101 versions of the ECDIS Chart 1. Please ask Jonathan for details

**Comment by HannuPeiponen** [comment]
> Should COLOUR **ADINF** be also considered for removal from S101PC ?

Colour "ADINF" is used by mariner symbol examples which are part of "ECDIS Chart 1". Jonathan Pritchard has created S-101 versions of the ECDIS Chart 1. Therefore, colour ADINF is needed in PC of S-101.

**Comment by DavidGrant-NIWC** [comment]
> I have noted that Jonathan Pritchard has created S-101 version of the "ECDIS Chart 1". Some of these sysmbols are used in "ECDIS Chart 1", therefore, such symbols should stay in PC of S-101 as the machine-readable S-100 compliant system needs them for rendering of the S-101 versions of the ECDIS Chart 1. Please ask Jonathan for details

@kusala9 could you identify the items which will be used in Chart 1? Is there an expectation that the PC will be updated to support display/use of these items?

**Comment by alvarosanuy** [comment]
These are the portrayal elements mentioned in S-52 LUT and registered in the GI Registry as 'Mariners' Objects'.
[Mariners - S-52 PL & GI Registry.docx](https://github.com/S-101-Portrayal-subWG/Working-Documents/files/15172059/Mariners.-.S-52.PL.GI.Registry.docx)

The other ones listed by Dave seem to exist in the GI Registry and are referenced in the S-52 PL DAI file (I checked some entries, not all).

- [ ] We should removed all the elements listed by Dave from the S-101PC **unless** required to support S-101 ECDIS Chart 1 implementation (awaiting feedback from @kusala9).

**Couple of exceptions to the above - Functional requirements not linked to mariners' preferences or ECDIS GUI requirements (e.g. North arrow or Scale Bar)**. Until the day we have an IEC Catalogue, we need to support S-101 functional requirements in ECS that may not support S-52.

- Does any other S-100 PS use the Overscale pattern (**OVERSC01**)? It is certainly used by S-101 datasets therefore I shouldn't remove it from the S-101PC.
 
- Similarly, **INDHLT01, INDHLT02 & DNGHILIT** are used to trigger Alerts & Indications Highlights in support of S-101 A&I. Consequently, they shouldn't be removed from the S-101 PC.

**Comment by DavidGrant-NIWC** [comment]
Recommend this is deferred to S-101 1.4.

**Comment by kusala9** [comment]
> > I have noted that Jonathan Pritchard has created S-101 version of the "ECDIS Chart 1". Some of these sysmbols are used in "ECDIS Chart 1", therefore, such symbols should stay in PC of S-101 as the machine-readable S-100 compliant system needs them for rendering of the S-101 versions of the ECDIS Chart 1. Please ask Jonathan for details
> 
> 
> 
> @kusala9 could you identify the items which will be used in Chart 1? Is there an expectation that the PC will be updated to support display/use of these items?

Apologies. I only caught sight of this this week. Yes, some of the symb ps are required for the proposed S-164 Chart 1. I will compile the list and post to this thread. Tom has proposed a paper for the PT to retain these symbols (and colors / lifestyles) in the 101 PC.

**Comment by DavidGrant-NIWC** [comment]
> Yes, some of the symbols are required for the proposed S-164 Chart 1

How are these going to be referenced? They aren't referenced by any existing rule and there aren't currently features in S-101 to associate with a new rule.

**Comment by alvarosanuy** [comment]
Deferring to PC 1.4.0 ...

**Comment by TomRichardson6** [comment]
Further to Jonathan's point above I will reference this issue in the paper to S-101PT13 in June. This proposes a new feature with an attribute similar to the SYMINS attribute present in S-57 ENC. So this may require some symbols and possibly even colors to be included in the Portrayal Catalogue without specific rules which reference on the basis that drawing instructions in this attribute may reference them. Agree to defer to 1.4.0. But due to timeframes it may be necessary to start drafting the associated rule and preparing a listing of symbols ahead of PT13 assuming that the paper is endorsed by PT13.

**Comment by DavidGrant-NIWC** [comment]
Suggest creation of an issue in the DCEG/PS github to track implementation of the attribute and associated requirements. I assume that the attribute would contain one or more DEF (see S-100 Part 13) formatted drawing instructions:
```
"PointInstruction:LIGHTS82"
"Dash:0,3.6;LineStyle:_simple_,5.4,1.28,OUTLW;LineInstruction:_simple_"
```

Note that this creates a potential attack vector for code injection techniques. The PC rule should impose some simple restrictions on the supported functionality. The PS should also strongly encourage OEM's Lua implementations to be sandboxed and to ensure datasets are properly signed.

**Comment by alvarosanuy** [comment]
@DavidGrant-NIWC - Just wondering if you have enough guidance on this issue as to proceed with PC 1.4.0 implementation.
If not, can you please detail what is needed from your point of view? The issue got a bit sidetracked by references to ECDIS Chart 1.
**My understanding is that** you would remove all the portrayal elements you listed in your comment from 30APR **less** those that may be required for ECDIS Chart 1 implementation (linked to JP's paper to S101PT13 ?).

**Comment by DavidGrant-NIWC** [comment]
@kusala9 has an initial implementation he will provide; he was just waiting on the 1.4 FC (I added the draft 1.4 FC yesterday)

**Comment by kusala9** [comment]
I will work with the 1.4 FC and list the symbols required. I have a simple LUA routine which accomplishes everything for Chart 1 (not sure we need anything more complex) and then I can list the symbols required. Rather than include the drawing instructions in their entirety it's easier to just provide either a symbol, colour token (for fill) or line style which seems to work fine. It is more restricted and so can be easily validated to avoid code injection issues. Is this to be done on a new issue, or documented on this one?

**Comment by DavidGrant-NIWC** [comment]
@alvarosanuy & @kusala9 confirm / update the list below. Items marked for retention are functional requirements or will be used in Chart 1.

### Area Fills
|AreaFill|Retain|Description|
|:---|:---:|:---|
|OVERSC01|Y|Overscale part of a display containing data from more than one navigation purpose (jail bars)|

### Line Styles
|LineStyle|Retain|Description|
|:---|:---:|:---|
|CHCRDEL1|Y|manual deletion|
|CHCRID01|Y|manual update|
|ERBLNA01|?|Electronic Bearing Line|
|INDHLT02|Y|mariners indication highlight|
|NONHODAT|Y|boundary of non-HO data|
|SCLBDY51|Y|chart scale boundary|

### Symbols
|Symbol|Retain|Description|
|:---|:---:|:---|
|BLKADJ01|Y|symbol for checking brightness and contrast|
|CHCRDEL1|Y|manually deleted or modified|
|CHCRID01|Y|manuallly updated|
|CHINFO08|Y|mariners information note|
|CHINFO09|Y|mariners caution note|
|CHINFO10|Y|manufacturers information note|
|CHINFO11|Y|manufacturers caution note|
|CHKSYM01|Y|symbol for checking symbol sizes|
|CURSRA01|?|ordinary cursor|
|CURSRB01|?|cursor with open centre|
|DNGHILIT|Y|transparent danger highlight for mariner's use|
|EBLVRM11|?|point of origin for an offset EBL or VRM|
|ERBLTIK1|?|range mark for an ERBL|
|INDHLT01|Y|indication highlight for mariner's use|
|NORTHAR1|Y|north arrow|
|REFPNT02|?|reference point, 'ghost cursor' (user interface)|
|SCALEB10|Y|one mile scale bar|
|SCALEB11|Y|10 mile latitude bar|
|TIDCUR01|?|predicted tidal stream or current direction|
|TIDCUR02|?|actual tidal stream or current direction|
|TIDCUR03|?|box for current strength|

**Comment by kusala9** [comment]
these are all the SY() lines in the current S-57 chart 1. From these I'll update the list above. I also need to go through the TX lines in the S-57 version and check nothing's been omitted.

- [ ]  AC(DEPDW);LS(SOLD,1,CHGRD)
- [ ]  AC(DEPVS);LS(SOLD,1,CHGRD)
- [ ]  AC(DNGHL,3);LS(SOLD,3,DNGHL) , 8 , O , MARINERS , 53010
- [ ]  AC(LANDA);LS(SOLD,1,CHGRD)
- [ ]  AC(NODTA);LS(SOLD,1,CHGRD)
- [ ]  AP(DIAMOND1)
- [ ]  AP(OVERSC01)
- [ ]  LC(CHCRDEL1)
- [ ]  LC(CHCRID01)
- [ ]  LC(CHRVDEL2)
- [ ]  LC(CHRVID02)
- [ ]  LC(INDHLT02) , 8 , O , MARINERS , 53010
- [ ]  LC(NONHODAT)
- [ ]  LC(QUESMRK1)
- [ ]  LC(SCLBDY51)
- [ ]  LS(SOLD,1,CHGRD)
- [ ]  LS(SOLD,3,ADINF)
- [ ]  LS(SOLD,3,DEPSC)
- [ ]  LS(SOLD,3,DNGHL) , 8 , O , MARINERS , 53010
- [ ]  LS(SOLD,3,NINFO)
- [ ]  LS(SOLD,3,RADLO)
- [ ]  LS(SOLD,3,RESBL)
- [ ]  LS(SOLD,3,TRFCF)
- [ ]  SY(BLKADJ01)
- [ ]  SY(BOYINB01)
- [ ]  SY(BRIDGE01)
- [ ]  SY(BRTHNO01);SY(BCNCAR01); TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);SY(BCNCAR01);TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);SY(BCNCAR02);TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);SY(BCNCAR03);TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);SY(BCNCAR04);TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);SY(BCNISD21);TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);SY(BOYLAT13);TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);SY(BOYLAT14);TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);SY(BOYLAT23);TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);SY(BOYLAT24);TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);SY(BOYSAW12);TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);SY(BOYSPP11); TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);SY(BOYSPP11);TX( V-AIS ,3,2,2, 15110 ,2,0,CHMGD,11)
- [ ]  SY(BRTHNO01);TE( %s , OBJNAM ,2,1,2, 15110 ,4,-1,CHMGD,29)
- [ ]  SY(CHCRDEL1)
- [ ]  SY(CHCRID01)
- [ ]  SY(CHDATD01)
- [ ]  SY(CHINFO08)
- [ ]  SY(CHINFO09)
- [ ]  SY(CHINFO10)
- [ ]  SY(CHINFO11)
- [ ]  SY(CHKSYM01)
- [ ]  SY(CHRVDEL1)
- [ ]  SY(CHRVID01)
- [ ]  SY(CTNARE51);LC(CTNARE51);TX(OBJNAM,1,2,3, 15110 ,0,0,CHBLK,26)
- [ ]  SY(CTNARE51);LC(CTNARE51);TX(PRELMN,1,2,3, 15110 ,0,0,CHBLK,26)
- [ ]  SY(DANGER02)
- [ ]  SY(DANGER03)
- [ ]  SY(DNGHILIT)
- [ ]  SY(FAIRWY51)
- [ ]  SY(FOULGND1)
- [ ]  SY(FSHFAC02)
- [ ]  SY(FSHFAC03)
- [ ]  SY(FSHGRD01)
- [ ]  SY(FSHHAV01)
- [ ]  SY(HILTOP11)
- [ ]  SY(INDHLT01)
- [ ]  SY(INFORM01)
- [ ]  SY(ISODGR01)
- [ ]  SY(LNDARE01)
- [ ]  SY(LOWACC01)
- [ ]  SY(MARCUL02)
- [ ]  SY(NORTHAR1)
- [ ]  SY(OBSTRN01)
- [ ]  SY(OBSTRN02)
- [ ]  SY(OBSTRN03)
- [ ]  SY(OBSTRN11)
- [ ]  SY(OFSPLF01)
- [ ]  SY(QUAPOS01)
- [ ]  SY(QUESMRK1)
- [ ]  SY(SCALEB10)
- [ ]  SY(SCALEB11)
- [ ]  SY(SNDWAV02)
- [ ]  SY(SPRING02)
- [ ]  SY(TMBYRD01)
- [ ]  SY(TSLDEF51)
- [ ]  SY(TWRDEF51)
- [ ]  SY(TWRTPT52)
- [ ]  SY(UWTROC03)
- [ ]  SY(UWTROC04)
- [ ]  SY(WEDKLP03)
- [ ]  SY(WRECKS01)
- [ ]  SY(WRECKS04)
- [ ]  SY(WRECKS05)

**Comment by kusala9** [comment]
and also check the other symbols are still included/used in the PC.

**Comment by alvarosanuy** [comment]
**Decision at PsWG meeting on 10/7/2024**

Retain all symbols until further notice. Close this issue.
